### PR TITLE
Add stop api to rpc agent

### PR DESCRIPTION
**Pull request by xush6528**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#28236 Add stop api to rpc agent**

`

Differential Revision: [D5516149](https://our.internmc.facebook.com/intern/diff/D5516149/)

### CODE REVIEW

File: torch/csrc/distributed/rpc/process_group_agent.cpp [hunk 1]
```diff
@@ -254,12 +254,14 @@ void ProcessGroupAgent::sync() {
 }
 
 void ProcessGroupAgent::start() {
+  start_ = true;
```
Comment (mrshenli) [inline torch/csrc/distributed/rpc/process_group_agent.cpp]: Doe this mean we can call start()/stop() multiple times on the same ProcessGroupAgent instance?

**Review by mrshenli — APPROVED** [review]

**Comment by pietern** [comment]
@xush6528 This is a dup of #28235 and #28238. Can you close the ones that are no longer relevant?

File: torch/csrc/distributed/rpc/process_group_agent.cpp (hunk 1 above)
Comment (xush6528) [reply torch/csrc/distributed/rpc/process_group_agent.cpp]: Yes, and this requires clean shutdown. All statefule variables should be put in the correct state.

**Comment by xush6528** [comment]
Duplicate of #28238

**Pull request closed without merging**